### ISSUE TITLE
'new layout' Facebook notifications menu doesn't receive mouse wheel scroll events

### ISSUE THREAD

**Issue by filbo** [Christoph142/modern-scroll#104]
Hi,

This probably can't be reproduced without a Facebook account, so I hope one is available.

On Facebook's recently replaced desktop web UI (the default if you go to `www.facebook.com` from a desktop), the Notifications menu does not scroll with modern-scroll active.

This is the most direct repro (assuming FB account, sufficiently connected to have more than a screen's worth of notifications):

1. Go to facebook.com/notifications (*1); the notifications menu appears in the right column and **is** scrollable
2. Click on a post notification within the menu; the menu remains open, while the selected post opens in the left part of the page
3. Use mouse wheel to scroll the post (left part of page); this should work
4. Use mouse wheel to scroll the still-open notifications menu (right part of page); this _should_ work, but does not.  Instead, the post (left part of page) scrolls despite the mouse pointer being on top of the menu

It is as if modern-scroll is routing the scroll events to the wrong layer of the page.  I don't know if that means z-index, frames, or what -- did not try to decompose the page internals.  I normally expect a different sort of scroll-cascading behavior, in which the pointed-at object scrolls, but when it reaches the end of its scrollable area, then and only then, scroll events bubble outwards to the parent 'layer'.  (I'm not sure I'm 100% in favor of that default behavior, but it is the long-observed default which is being violated here; in the direction of _always_ bubbling outwards.)

Particularly note that scrolling the notifications _does_ work initially; it is not until one has been clicked through, instantiating a post in the left part of the page, that it stops working.

(*1) Note that this relates to a new behavior of the FB 'new layout'.  In the old layout, clicking an item in the notifications menu replaced the entire page contents; in the new, it puts a post into the left part of the page (as if an iframe, though I think it is not).  This behavior happens only when opening facebook.com/notifications, **_not_** when merely pulling down the notifications menu from an existing FB page.

Possibly relevant details:

Linux Mint 17.3 (!old!) x86_64
Linux kernel: 4.4.0-119-generic #143~14.04.1-Ubuntu SMP Mon Apr 2 18:04:36 UTC 2018 x86_64
X server: X.org 1.17.1
Window manager: marco 1.12.0
Session: MATE 1.12.0
Browser: Opera 72.0.3791.0 (opera-developer branch)

This is not intended as a 'delta report' suggesting that this worked correctly on FB's 'old layout' or anywhere else.  I just started using modern-scroll today, I have no prior experience with it.  I see that it works in many places, including internal scrolling areas of many pages; and indeed on the problem page, until clicking a particular post.  There are probably other places on FB (and elsewhere) with the same behavior, which I haven't noticed yet.  I merely pinpoint one specific case that I _know_ how to reproduce.

It might be modern-scroll-settings related.  I did not find a way of saving settings to a text file.

It does _not_ relate to the 'Style all scrollbars' setting -- happens the same whether this is on or off.

**Comment by filbo** [comment]
Unrelated, and I will open a separate issue on this when I have time, but: the bookmarks / headings setting makes it difficult to use the modern-scroll options within its own menu, pulled down from the browser toolbar menu button.  The option checkboxes are near the right edge (I was fighting with the 'Save' icon).  Could probably workaround by using left-toolbar, but I don't want to.

I'm vaguely thinking of fixes like: longer delay before popping up bookmarks; delay doesn't start until mouse is actually still (not moving, within some small jitter delta) in the target area; a way to dismiss the bookmarks (like, waggle the mouse left-right-left-right in that area while they're up -- with motions large and sharp enough to be intentional; stay dismissed until mouse has left and reentered the area).

On further testing as I write this, I see that the bookmarks only come up if I traverse the scrollbar itself, so if I am very careful to only approach it but not cross into it, it's OK.  But I am not habitually that careful, my typical mouse approach probably looks like a big curve which somewhat overshoots the target, then pulls back.  So a simple hover delay before raising the bookmarks would help a great deal.

**Comment by filbo** [comment]
Is there anything I can do to move this forward?  I really like modern-scroll but its failure to cascade scrolling events is making it hard to use.

**Comment by Christoph142** [comment]
This works fine for me. Both parts of the page scroll like they should when hovered. Any additional mods, extensions,...?

**Comment by filbo** [comment]
Only 9, including plenty invasive stuff.  So I disabled **all** of them, loaded a new facebook.com/notifications tab, instantly reproduced the problem.

To make it go wrong, you need to actually click on one of the notifications.  The n-menu should remain open (because of where you started).  Once clicking through a notification, the n-menu become anchored.

Since it's a bug, it doesn't work perfectly :)  I've occasionally had it not happen, and I've occasionally had it break loose from the problem state.  But it 'works' 90+% of the time.  Try a few times?

**Comment by filbo** [comment]
Is there a way to save modern scroll settings to an external file so I can more easily run through everything I have set, see if it's relevant?

**Comment by filbo** [comment]
I saved internally, reset to Default, and the problem is gone.  So it does relate to some ms setting; but it's hard to pick out specifically what I've changed without a settings file to look at.

**Comment by filbo** [comment]
...and, toggling on 'Scrolling functions > for the mouse' (at its default settings of 100%, 100%) reproduces the problem.

Yay!  Now I can have ms enabled all the time, only 1% impaired.
